nidirect_related_content/src/RelatedContentManager.php

```diff
@@ -0,0 +1,314 @@
+ *
+ * @package Drupal\nidirect_related_content
+ */
+class RelatedContentManager {
+
+  public const CONTENT_ALL = 'all';
+  public const CONTENT_THEMES = 'themes';
+  public const CONTENT_NODES = 'nodes';
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current route.
+   *
+   * @var \Drupal\Core\Routing\CurrentRouteMatch
+   */
+  protected $routeMatch;
+
+  /**
+   * Drupal renderer service.
+   *
+   * @var \Drupal\Core\Render\Renderer
+   */
+  protected $renderer;
+
+  /**
+   * Theme/term ids.
+   *
+   * @var array
+   *   Array of taxonomy term ids.
+   */
+  protected $termIds;
+
+  /**
+   * Theme content.
+   *
+   * @var array
+   *   Array of theme content.
+   */
+  protected $content;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity query manager.
+   * @param \Drupal\Core\Routing\CurrentRouteMatch $route_match
+   *   Current route match.
+   * @param \Drupal\Core\Render\Renderer $renderer
+   *   Drupal renderer.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, CurrentRouteMatch $route_match, Renderer $renderer) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->routeMatch = $route_match;
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * Fetches content (Terms & Nodes) for a theme.
+   *
+   * @param array $term_ids
+   *   Array of taxonomy term ids to retrieve content for.
+   * @param string $content
+   *   Return themes, nodes or both.
+   *
+   * @return $this
+   */
+  public function getThemeContent(array $term_ids = NULL, $content = self::CONTENT_ALL): RelatedContentManager {
+
+    // If no terms_ids are passed in try and extract from the current request,
+    // either a node or taxonomy page.
+    if ($term_ids === NULL) {
+      $route_name = $this->routeMatch->getRouteName();
+
+      if ($route_name === 'entity.node.canonical') {
+        $node = $this->routeMatch->getParameter('node');
+        if ($node->hasField('field_subtheme') && !$node->get('field_subtheme')->isEmpty()) {
+          $term_ids[] = $node->get('field_subtheme')->getString();
+        }
+        if ($node->hasField('field_site_themes') && !$node->get('field_site_themes')->isEmpty()) {
+          $term_ids[] = $node->get('field_site_themes')->getString();
+        }
+      }
+      elseif ($route_name === 'entity.taxonomy_term.canonical') {
+        $term = $this->routeMatch->getParameter('taxonomy_term');
+        $term_ids[] = $term->id();
+        if ($term->hasField('field_supplementary_parents') && !$term->get('field_supplementary_parents')->isEmpty()) {
+          $term_ids[] = $term->get('field_supplementary_parents')->getString();
+        }
+      }
+      else {
+        return $this;
+      }
+    }
+
+    $this->setTerms($term_ids);
+
+    if ($content === 'themes') {
```

Comment:
Worth referencing the class constants here instead of literal string values? Eg: `$this-> CONTENT_THEMES` instead of `themes`

Reply:
Doh,  very true.